mmcv/utils/path.py

```diff
@@ -69,8 +69,10 @@ def _scandir(dir_path, suffix, recursive):
                     yield rel_path
             else:
                 if recursive:
-                    yield from _scandir(
-                        entry.path, suffix=suffix, recursive=recursive)
+                    if os.path.isdir(entry.path):
```

Comment:
```python
if recursive and os.path.isdir(entry.path):
    yield from _scandir(entry.path, suffix=suffix, recursive=recursive)
else:
    continue
```

Reply:
I haven't tried it but the logic seems to be the same.

Reply:
yet, we can rewrite it to make it simpler.

Reply:
and it is better to add a unittest for this